distribution/lib/Standard/Database/0.0.0-dev/src/Data/Column.enso

```diff
@@ -623,22 +626,74 @@ type Column
              example_round = Examples.decimal_column.round 2
     round : Integer -> Boolean -> Column | Illegal_Argument | Invalid_Value_Type
     round self decimal_places=0 use_bankers=False =
-        Decimal_Places_Helpers.check_decimal_places decimal_places <| Value_Type.expect_numeric self <|
-            new_name = self.naming_helpers.function_name "round" [self]
-            scale = 10 ^ decimal_places
-            scaled = self * scale
-            round_base = scaled.floor
-            round_midpoint = (round_base + 0.5) / scale
-            even_is_up = (self >= 0).iif ((scaled.truncate % 2) != 0) ((scaled.truncate % 2) == 0)
-            half_goes_up = if use_bankers then even_is_up else (self.const True)
-            do_round_up = half_goes_up.iif (self >= round_midpoint) (self > round_midpoint)
-            result_float = do_round_up.iif ((round_base + 1.0) / scale) (round_base / scale)
-
-            should_cast_to_integer = decimal_places <= 0 || self.value_type.is_integer
-            result = if should_cast_to_integer.not then result_float else
-                result_float.cast Value_Type.Integer
-
-            result.rename new_name
+        Decimal_Places_Helpers.check_decimal_places decimal_places <|
+            k = case self.value_type of
+                    Value_Type.Integer _ -> self.round_integer decimal_places use_bankers
+                    Value_Type.Float _ -> self.round_float decimal_places use_bankers
+                    Value_Type.Decimal _ _ -> self.round_float decimal_places use_bankers
+                    _ ->
+                        expected = "Value_Type.Decimal, Value_Type.Float, or Value_Type.Integer"
+                        Error.throw (Invalid_Value_Type.Column expected self.value_type self.name)
+            #IO.println 'k'
+            #IO.println k
+            k
```

Comment:
Just a reminder to remove the debug logs before merge

Reply:
Done.